PR: https://github.com/odoo/odoo/pull/160169

From: 02f756d061f1e52a2265e051ac57b13b311fb115
From: David Monnom (moda)
Date: 2024-04-29 23:21:31

Breaking data model changes score: 2, change matches:
-    _inherit = 'pos.order'
-    _inherit = 'pos.session'

Total Changes: 34

[IMP] pos_*: allow arbitrary data in related models

*: All pos related modules

This commit allows to access arbitrary data in related models. This is
useful for data that are computed on the server side and that are not
recomputed on the client side.

Before this commit, it was only possible to access data that are linked
to a field in the model.

Now custom data coming from the server can be added to any record with
the prefix `_` in the key. For example, if the server sends the
following data for `pos.order`:

Classic fields:
```
{
  id: 1,
  name: 'Order 1',
  partner_id: 1,
  partner_name: 'Partner 1',
}
```

Custom fields:
```
{
  id: 1,
  name: 'Order 1',
  partner_id: 1,
  partner_name: 'Partner 1',
  _custom_field: 'Custom value',
}
```

closes odoo/odoo#160169

Related: odoo/enterprise#60017
Signed-off-by: Adrien Guilliams (adgu) <adgu@odoo.com>

================================= pseudo patch: =================================

--- a/addons/pos_razorpay/models/pos_payment.py
+++ b/addons/pos_razorpay/models/pos_payment.py
@@ -1,4 +1,4 @@
-from odoo import api, models, fields
+from odoo import models, fields
 
 class PosPayment(models.Model):
 
@@ -12,22 +12,3 @@ class PosPayment(models.Model):
     razorpay_reverse_ref_no = fields.Char('Razorpay Reverse Reference No.')
     razorpay_card_scheme = fields.Char('Razorpay Card Scheme')
     razorpay_card_owner_name = fields.Char('Razorpay Card Owner Name')
-
-class PosOrder(models.Model):
-
-    _inherit = 'pos.order'
-
-    @api.model
-    def _payment_fields(self, order, ui_paymentline):
-        payment_fields = super()._payment_fields(order, ui_paymentline)
-        payment_fields.update({
-                'razorpay_authcode': ui_paymentline.get('razorpay_authcode'),
-                'razorpay_issuer_card_no': ui_paymentline.get('razorpay_issuer_card_no'),
-                'razorpay_issuer_bank': ui_paymentline.get('razorpay_issuer_bank'),
-                'razorpay_payment_method': ui_paymentline.get('razorpay_payment_method'),
-                'razorpay_reference_no': ui_paymentline.get('razorpay_reference_no'),
-                'razorpay_reverse_ref_no': ui_paymentline.get('razorpay_reverse_ref_no'),
-                'razorpay_card_scheme': ui_paymentline.get('razorpay_card_scheme'),
-                'razorpay_card_owner_name': ui_paymentline.get('razorpay_card_owner_name'),
-            })
-        return payment_fields

--- a/addons/pos_razorpay/models/pos_session.py
+++ b/None
@@ -1,13 +0,0 @@
-# -*- coding: utf-8 -*-
-# Part of Odoo. See LICENSE file for full copyright and licensing details.
-
-from odoo import models
-
-class PosSession(models.Model):
-    _inherit = 'pos.session'
-
-    def _load_data_params(self, config_id):
-        params = super()._load_data_params(config_id)
-        params['pos.payment']['fields'] + ['razorpay_authcode', 'razorpay_issuer_card_no', 'razorpay_issuer_bank',
-            'razorpay_payment_method', 'razorpay_reference_no', 'razorpay_reverse_ref_no', 'razorpay_card_scheme', 'razorpay_card_owner_name']
-        return params
